datasafe-storage/datasafe-storage-impl-db/src/main/java/de/adorsys/datasafe/storage/impl/db/DatabaseConnectionRegistry.java

```diff
@@ -106,30 +116,46 @@ protected DatabaseCredentials getCredentials(AbsoluteLocation location) {
         URI uri = location.location().asURI();
         String userInfo = uri.getUserInfo();
 
-        if (null !=  userInfo && !"".equals(userInfo)) {
+        if (null != userInfo && !"".equals(userInfo)) {
             return new DatabaseCredentials(location);
         }
 
         return providedCredentials.entrySet().stream()
-                .filter(it -> uri.toASCIIString().startsWith(it.getKey()))
+                .filter(compareDatabaseURI(uri))
```

Comment:
Why this extractor is needed? Matching must be done using key

Reply:
for reduce code in chain

Reply:
sorry, i meant, it -> uri.toASCIIString().startsWith(it.getKey()) was ok, what was the purpose of its removal?

Reply:
because the compare code is bigger than was before

Reply:
https://github.com/adorsys/datasafe/pull/134/files/bc4f7d9161e91e9c54827bdf4077968de27264fd#diff-30a79ccf87b868fb7db1633b7bc742b5R131

Reply:
done